task/host-sp-comms/src/main.rs

```diff
@@ -701,13 +710,24 @@ impl ServerImpl {
 
         // We defer any actions until after we've serialized our response to
         // avoid borrow checker issues with calling methods on `self`.
+        //
+        // NOTE: setting any action will also clear the `host_boot_just_failed`
+        // flag, to avoid keeping incorrect state around. Since we have no
+        // explicit way of tracking the host boot software's lifecycle, this is
+        // a compromise.
         let mut action = None;
+
         let response = match request {
             HostToSp::_Unused => {
                 Some(SpToHost::DecodeFailure(DecodeFailureReason::Deserialize))
             }
             HostToSp::RequestReboot => {
-                action = Some(Action::RebootHost);
+                if core::mem::take(&mut self.host_boot_just_failed) {
```

Comment:
The use of `mem::take` here is very cute! But, IIUC, since we clear the flag later if `action` is `Some`, we don't actually _need_ to clear it when we read the flag's value here, do we? Since both branches set `action` to `Some` regardless?

Up to you whether this actually ought to change, but at a first glance it kind of appeared that this was the only case where we clear the flag, so it was a bit less obvious that _any_ branch where `action` is `Some` clears the flag (despite the comment). :woman_shrugging:

Reply:
Yeah, I wound up doing both here to make it really really unlikely that we forget to clear the flag. Arguably unnecessary. What do you think would be clearest?

Reply:
I think maybe leaving it as is, but commenting that this also gets cleared because we've set `action = Some(...)`? Honestly, I think it's not really a big deal --- mostly, I wanted to make sure my understanding was correct.

Reply:
The fact that it threw you suggests that I need to make it more clear, so I'll see about adding some comments.